tutorials/tensordictmodule.ipynb

```diff
@@ -0,0 +1,1060 @@
+   "cell_type": "code",
+   "execution_count": 12,
+   "id": "9f25a6ba-c9f5-4eea-8e4b-c4f456d19b4c",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "TensorDict(\n",
+      "    fields={\n",
+      "        hidden: Tensor(torch.Size([3, 8]), dtype=torch.float32),\n",
+      "        input: Tensor(torch.Size([3, 4]), dtype=torch.float32)},\n",
+      "    batch_size=torch.Size([3]),\n",
+      "    device=cpu,\n",
+      "    is_shared=False)\n",
+      "TensorDict(\n",
+      "    fields={\n",
+      "        action: Tensor(torch.Size([3, 4]), dtype=torch.float32),\n",
+      "        hidden: Tensor(torch.Size([3, 8]), dtype=torch.float32),\n",
+      "        input: Tensor(torch.Size([3, 4]), dtype=torch.float32),\n",
+      "        loc: Tensor(torch.Size([3, 4]), dtype=torch.float32),\n",
+      "        sample_log_prob: Tensor(torch.Size([3, 1]), dtype=torch.float32),\n",
+      "        scale: Tensor(torch.Size([3, 4]), dtype=torch.float32)},\n",
+      "    batch_size=torch.Size([3]),\n",
+      "    device=cpu,\n",
+      "    is_shared=False)\n"
+     ]
+    }
+   ],
+   "source": [
+    "from torchrl.modules import ProbabilisticTensorDictModule\n",
+    "from torchrl.modules import  TanhNormal, NormalParamWrapper\n",
+    "import functorch\n",
+    "td = TensorDict({\"input\": torch.randn(3, 4), \"hidden\": torch.randn(3, 8)}, [3,])\n",
+    "net = NormalParamWrapper(torch.nn.GRUCell(4, 8))\n",
+    "module = TensorDictModule(net, in_keys=[\"input\", \"hidden\"], out_keys=[\"loc\", \"scale\"])\n",
+    "td_module = ProbabilisticTensorDictModule(\n",
+    "   module=module,\n",
+    "   dist_param_keys=[\"loc\", \"scale\"],\n",
+    "   out_key_sample=[\"action\"],\n",
+    "   distribution_class=TanhNormal,\n",
+    "   return_log_prob=True,\n",
+    "   )\n",
+    "print(td)\n",
+    "td_module(td)\n",
+    "print(td)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "406b1caa-bcec-4317-b685-10df23352154",
+   "metadata": {},
+   "source": [
+    "### `Actor`"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e139de7d-0250-49c0-b495-8b5a404821f5",
+   "metadata": {},
+   "source": [
+    "Actor inherits from `TensorDictModule` and comes with a default value for `out_keys` of `[\"action\"]`.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cceeade9-47f1-4e92-897a-dd226c9371a6",
+   "metadata": {},
+   "source": [
+    "### `ProbabilisticActor`"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4fd0f53e-90aa-49a9-9d8f-5a260255e556",
+   "metadata": {},
+   "source": [
+    "General class for probabilistic actors in RL that inherits from `ProbabilisticTensorDictModule`.\n",
+    "Similarly to `Actor`, it comes with default values for the `out_keys` (`[\"action\"]`).\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "dbd48bb2-b93b-4766-b7a7-19d500f17e2d",
+   "metadata": {},
+   "source": [
+    "### `ActorCriticOperator`"
```

Comment:
Show the various ActorCritic and ActorValue operators, show what they do, what the differences are.
Also, ValueOperator is missing.